### DISCUSSION TITLE
Can next/image respect full names after optimization?

### DISCUSSION THREAD

**Discussion by glancer000** [vercel/next.js#28416] [Help]
Is there a way for the optimized images from next/image respect the original file name?

> **Create good URL structure for your images:** Google uses the URL path as well as the file name to help it understand your images. Consider organizing your image content so that URLs are constructed logically. ([Source](https://developers.google.com/search/docs/advanced/guidelines/google-images))

Right now, my optimized images get random URLs like
https://www.oncorehq.com/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Foncore%2Fimage%2Fupload%2Fv1629415317%2FOncore_sales_topic_015f3f3b36.png&w=1080&q=75

**Comment by leerob** [comment]
Thanks for reporting! You are correct, this currently doesn't exist. But it is good feedback I will forward to the team 👍

**Reply by glancer000** [reply]
Thanks for passing the information along. Is there a way to get updates on the status of this feedback and if it gets incorporated into an update?

**Reply by leerob** [reply]
When you say it's impacting SEO, are you talking about Image Search specifically?

**Reply by glancer000** [reply]
It will definitely impact Image Search but could have implications for regular SEO as well. When Google crawls your page, it is looking for signals that indicate relevancy and context to the topic you are trying to rank for. The clearer you can give those signals, the easier it is for Google to understand your page and rank it accordingly. Below is a snippet that hits on that a bit.

"Google extracts information about the subject matter of the image from the content of the page, including captions and image titles. Wherever possible, make sure images are placed near relevant text and on pages that are relevant to the image subject matter.

Likewise, the filename can give Google clues about the subject matter of the image. For example, my-new-black-kitten.jpg is better than IMG00023.JPG. If you localize your images, make sure you translate the filenames, too."

So from what I understand, it will directly impact Image Search but it may also take away one additional signal to Google to help give them context for regular rankings.
